core/Tracker/TrackerCodeGenerator.php

```diff
@@ -93,6 +115,7 @@ public function generate($idSite, $piwikUrl, $mergeSubdomains = false, $groupPag
 
         $codeImpl = array(
             'idSite'                  => $idSite,
+            // TODO why sanitizeInputValue() and not json_encode?
             'piwikUrl'                => Common::sanitizeInputValue($piwikUrl),
```

Comment:
If you mean doing something like: `'"' . json_encode($protocol) . json_encode($url) . '"'` won't it create a string like: `'""http://""path""'`? Ie, w/ the protocol and url surrounded w/ quotes.

What I meant was that to do it properly, you'd have to concat the protocol + url first, then do the json_encode (w/o the surrounding quotes of course).